In [10]:
#Task-1
import pandas as pd 

df = pd.read_csv("./data/orders.csv", index_col = "order_id")

print("\nFirst five rows:\n", df.head())

print("\nLast five rows:\n", df.tail())



First five rows:
           customer_name   product  quantity  unit_price  order_date     status
order_id                                                                      
1001      Alice Johnson    Laptop       1.0     85000.0  2024-01-15  delivered
1002          Bob Smith     Mouse       3.0       499.0  2024-01-16    pending
1003        Carol White   Monitor       2.0     15000.0         NaN  delivered
1004         Dave Brown  Keyboard       NaN      1200.0  2024-01-18  cancelled
1005          Eva Green    Laptop       1.0     85000.0  2024-01-19  delivered

Last five rows:
          customer_name   product  quantity  unit_price  order_date     status
order_id                                                                     
1016       Paul Harris     Mouse       6.0       499.0  2024-01-30    pending
1017      Quinn Martin   Monitor       NaN     14500.0  2024-01-31  delivered
1018       Rachel King    Laptop       1.0     85000.0         NaN  delivered
1019        Sam Wrig

In [ ]:
#Task-2
print("Structural inspection of the dataset:\n", df.info())


<class 'pandas.core.frame.DataFrame'>
Index: 20 entries, 1001 to 1020
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   customer_name  20 non-null     object 
 1   product        20 non-null     object 
 2   quantity       16 non-null     float64
 3   unit_price     20 non-null     float64
 4   order_date     16 non-null     object 
 5   status         20 non-null     object 
dtypes: float64(2), object(4)
memory usage: 1.1+ KB
Structural inspection of the dataset:
 None


Q1:How many rows and columns does the dataset have?
Ans: 20 rows and 6 columns

Q2: Which columns have missing values and how many nulls does each have?
Ans: The following columns have null values:
    quantity : 4 null
    order_date : 4 null 

Q3: Which column needs a data type conversion before any date-based analysis, and what function would you use?
Ans: order_date need type conversion from object to date using : df["order_date"] = pd.to_datetime(df["order_date"])

In [12]:
#Task-3
print("Statistical summary of the dataset:\n", df.describe())

Statistical summary of the dataset:
         quantity    unit_price
count  16.000000     20.000000
mean    2.437500  20224.600000
std     1.504161  33573.389607
min     1.000000    499.000000
25%     1.000000    799.000000
50%     2.000000   2500.000000
75%     3.000000  14625.000000
max     6.000000  85000.000000


Q1: What is the median unit_price?
Ans: Median unit_price = 2500.0000

Q2: Does unit_price appear to be skewed? Explain using values from the output.
Ans: Yes, unit_price is right skewed as mean (20224.6) is signifiantly greater than median (2500).